### PR TITLE
Shell linter

### PR DESCRIPTION
**Pull request by anniepamela2020**
Edit and lint shell files using shellcheck

Fixes the syntax in the shell scripts

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -90,3 +90,7 @@ lint-yaml: ## Lint yaml files
 	cd resources && yamllint .
 	cd examples && yamllint .
 	cd platform && yamllint . 
+
+.PHONY: shell-linter
+shell-linter: ## Lint shell files
+	shellcheck "**/*.sh" "#docs" "#platform" "#scripts"
```
Comment (bradbeck) [inline Makefile:96]:
```suggestion
	shellcheck "**/*.sh" "#docs" "#platform" "#scripts"

```

File: platform/00-kubernetes-minikube-setup.sh [hunk 2]
```diff
@@ -14,7 +14,7 @@ MINIKUBE_SHA256=3bc218476cf205acf11b078d45210a4882e136d24a3cbb7d8d645408e423b8fe
 HELM_VERSION=v3.7.1
 HELM_FILE_NAME=helm-v3.7.1-linux-amd64.tar.gz
 HELM_URL=https://get.helm.sh/$HELM_FILE_NAME
-HELM_SHA256=6cd6cad4b97e10c33c978ff3ac97bb42b68f79766f1d2284cfd62ec04cd177f4
+"HELM_SHA256=6cd6cad4b97e10c33c978ff3ac97bb42b68f79766f1d2284cfd62ec04cd177f4"
```
Comment (bradbeck) [inline platform/00-kubernetes-minikube-setup.sh:17]:
```suggestion
HELM_SHA256="6cd6cad4b97e10c33c978ff3ac97bb42b68f79766f1d2284cfd62ec04cd177f4"
```

File: docs/bootstrap.sh [hunk 3]
```diff
@@ -3,7 +3,7 @@ set -euo pipefail
 
 # Define variables.
 C_GREEN='\033[32m'
-C_YELLOW='\033[33m'
+# C_YELLOW='\033[33m'
```
Comment (bradbeck) [inline docs/bootstrap.sh:6]:
```suggestion
```

File: platform/10-tekton-setup.sh [hunk 4]
```diff
@@ -6,7 +6,7 @@ GIT_ROOT=$(git rev-parse --show-toplevel)
 # Define variables.
 C_GREEN='\033[32m'
 C_YELLOW='\033[33m'
-C_RED='\033[31m'
+#C_RED='\033[31m'
```
Comment (bradbeck) [inline platform/10-tekton-setup.sh:9]:
```suggestion
```

File: platform/20-spire-setup.sh [hunk 5]
```diff
@@ -5,31 +5,31 @@ GIT_ROOT=$(git rev-parse --show-toplevel)
 QUICKSTART_DIR=$GIT_ROOT/platform/vendor/spire/quickstart
 
 # Define variables.
-C_GREEN='\033[32m'
+#C_GREEN='\033[32m'
```
Comment (bradbeck) [inline platform/20-spire-setup.sh:8]:
```suggestion
```

File: platform/20-spire-setup.sh [hunk 6]
```diff
@@ -5,31 +5,31 @@ GIT_ROOT=$(git rev-parse --show-toplevel)
...
 C_YELLOW='\033[33m'
-C_RED='\033[31m'
+#C_RED='\033[31m'
```
Comment (bradbeck) [inline platform/20-spire-setup.sh:10]:
```suggestion
```

File: scripts/provenance.sh [hunk 7]
```diff
@@ -8,16 +8,16 @@ else
...
+kubectl get taskrun "$TASKRUN" -o=json | jq > "$TASKRUN_JSON"
 jq \
   -r ".metadata.annotations[\"chains.tekton.dev/payload-taskrun-$TASKRUN_UID\"]" \
-  $TASKRUN_JSON  \
+  "$TASKRUN_JSON"  \
   | base64 --decode > payload.json
 jq \
   -r ".metadata.annotations[\"chains.tekton.dev/signature-taskrun-$TASKRUN_UID\"]" \
-  $TASKRUN_JSON > signature.pub
+  "$TASKRUN_JSON "> signature.pub
```
Comment (bradbeck) [inline scripts/provenance.sh:20]:
```suggestion
  "$TASKRUN_JSON" > signature.pub
```

File: platform/20-spire-setup.sh [hunk 8]
```diff
@@ -5,31 +5,31 @@ GIT_ROOT=$(git rev-parse --show-toplevel)
...
 # $1: namespace, $2: app label
 wait_for_pods () {
   echo -e "${C_YELLOW}Waiting: $2 pods in $1...${C_RESET_ALL}"
-  kubectl wait --timeout=5m --for=condition=ready pods -l app=$2 -n $1
+  kubectl wait --timeout=5m --for=condition=ready pods -l app="$2" -n "$1"
 }
 
 spire_apply() {
-  if [ $# -lt 2 -o "$1" != "-spiffeID" ]; then
+  if [ $# -lt 2 ] || [ $# -lt "$1" ] || [ $# != "-spiffeID" ]; then
```
Comment (bradbeck) [inline platform/20-spire-setup.sh:21]:
```suggestion
  if [ $# -lt 2 ] || [ "$1" != "-spiffeID" ]; then
```

File: scripts/gen-keys.sh [hunk 9]
```diff
@@ -21,7 +21,7 @@ set -ea
 
 export NAMESPACE=tekton-chains
 export SECRET_NAME=signing-secrets
-: ${COSIGN_PASSWORD:=""}
+: "${COSIGN_PASSWORD:=""}"
```
Comment (bradbeck) [inline scripts/gen-keys.sh:24]:
```suggestion
: "${COSIGN_PASSWORD:=}"
```

File: Makefile (hunk 1 above)
Comment (bradbeck) [inline Makefile:96]: This syntax doesn't appear to work for me.

File: Makefile [hunk 10]
```diff
@@ -90,3 +90,7 @@ lint-yaml: ## Lint yaml files
 	cd resources && yamllint .
 	cd examples && yamllint .
 	cd platform && yamllint . 
+
+.PHONY: shell-linter
+shell-linter: ## Lint shell files
```
Comment (rgreinho) [inline Makefile:95]: Should be "lint-shell" to be consistent with the other tasks.

File: docs/bootstrap.sh (hunk 3 above)
Comment (rgreinho) [inline docs/bootstrap.sh:6]: Remove it if it is not being used.

File: platform/10-tekton-setup.sh (hunk 4 above)
Comment (rgreinho) [inline platform/10-tekton-setup.sh:9]: Remove it if it is not being used.

File: platform/20-spire-setup.sh (hunk 6 above)
Comment (rgreinho) [inline platform/20-spire-setup.sh:10]: Remove them if they are not being used.

File: Makefile (hunk 1 above)
Comment (rgreinho) [inline Makefile:96]: Adding a meta-task calling all the lint tasks would be great.

**Review by rgreinho — CHANGES_REQUESTED** [review]
Good job overall 👍 

Just a few minor comments to address.

File: Makefile [hunk 11]
```diff
@@ -90,3 +90,7 @@ lint-yaml: ## Lint yaml files
 	cd resources && yamllint .
 	cd examples && yamllint .
 	cd platform && yamllint . 
+
+.PHONY: Lint-shell
+lint-shell: ## Lint shell files
+	shfmt -f ./ | xargs shellcheck
```
Comment (bradbeck) [inline Makefile:96]:
```suggestion
	shfmt -f ./ | xargs shellcheck

```

File: scripts/provenance.sh [hunk 12]
```diff
@@ -8,16 +8,16 @@ else
...
+kubectl get taskrun "$TASKRUN" -o=json | jq > "$TASKRUN_JSON"
 jq \
   -r ".metadata.annotations[\"chains.tekton.dev/payload-taskrun-$TASKRUN_UID\"]" \
-  $TASKRUN_JSON  \
+  "$TASKRUN_JSON"  \
   | base64 --decode > payload.json
 jq \
   -r ".metadata.annotations[\"chains.tekton.dev/signature-taskrun-$TASKRUN_UID\"]" \
-  $TASKRUN_JSON > signature.pub
+  "$TASKRUN_JSON"> signature.pub
```
Comment (bradbeck) [inline scripts/provenance.sh:20]:
```suggestion
  "$TASKRUN_JSON" > signature.pub
```

**Review by bradbeck — DISMISSED** [review]

File: Makefile [hunk 13]
```diff
@@ -80,7 +80,8 @@ docs-serve: ## Serve the site locally with hot-reloading
 .PHONY: docs-build
 docs-build: ## Build the documentation site
 	cd docs && zola build
-
+  
+  Task: linter-markdown lint-yaml lint-shell
```
Comment (bradbeck) [inline Makefile:84]:
```suggestion
.PHONY: linter
linter:  linter-markdown lint-yaml lint-shell ## Run all linters

```

**Review by bradbeck — APPROVED** [review]

**Comment by trmiller** [comment]
> Good job overall 👍
> 
> Just a few minor comments to address.

@rgreinho - are these addressed from your POV quite yet?

**Review by rgreinho — APPROVED** [review]

**Pull request merged**